lua/hop/treesitter.lua

```diff
@@ -0,0 +1,89 @@
+local jump_targets = require('hop.jump_target')
+local parsers = require('nvim-treesitter.parsers')
+
+local T = {}
+
+T.run = function()
+  ---@param opts Options
+  ---@return Locations
+  return function(opts)
+    local Locations = T.parse()
+    jump_targets.sort_indirect_jump_targets(Locations.indirect_jump_targets, opts)
+    return Locations
+  end
+end
+
+---
+---@param targets JumpTarget[]
+---@param row integer
+---@param col integer
+---@return boolean
+local function duplicate(targets, row, col)
+  for _, j in pairs(targets) do
+    if j.line == row and j.column == col + 1 then
+      return true
+    end
+  end
+  return false
+end
+
+---
+---@return Locations
+T.parse = function()
+  ---@type Locations
+  local locations = {
+    jump_targets = {},
+    indirect_jump_targets = {},
+  }
+
+  -- Check if buffer has a parser
+  local parser = parsers.get_parser()
```

Comment:
```suggestion
  local parser = vim.treesitter.get_parser()
```
plugins should not depend on nvim-treesitter.